### PR TITLE
[public-api] Add experimental TeamsService

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Adds protobuf definitions for experimental TeamsService

## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/14279

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/public-api/gitpod/experimental/v1/teams.proto [hunk 1]
```diff
@@ -59,3 +74,53 @@ message CreateTeamRequest {
...
+    TeamInvitation team_invitation = 1;
+}
+
+message UpdateTeamMemberRequest {
+    // team_id is the ID of the team in which the role is to be updated
+    string team_id = 1;
+
+    // team_member is the team member being updated.
+    TeamMember team_member = 2;
+
```
Comment (andrew-farries) [inline components/public-api/gitpod/experimental/v1/teams.proto:111]: nit: extra blank line here.

File: components/public-api/gitpod/experimental/v1/teams.proto [hunk 2]
```diff
@@ -59,3 +74,53 @@ message CreateTeamRequest {
...
+message DeleteTeamMemberRequest {
+    // team_id is the ID of the team in which a member should be deleted.
+    string team_id = 1;
+
+    // team_member_id is the ID of the TeamMember that should be deleted from the team.
+    string team_member_id = 2;
+}
+
+message DeleteTeamMemberResponse {}
```
Comment (andrew-farries) [inline components/public-api/gitpod/experimental/v1/teams.proto:126]: Does this need indication of success/failure?

**Review by andrew-farries — APPROVED** [review]
/hold for one nit and a question

File: components/public-api/gitpod/experimental/v1/teams.proto (hunk 2 above)
Comment (easyCZ) [reply components/public-api/gitpod/experimental/v1/teams.proto:126]: It's indicated through the notion of error resulting from the call (status code). It succeeds if there's no error.
We may want to add more details about the delete here but so far I've not found the use-case and server also returns nil.

Comment (easyCZ) [reply components/public-api/gitpod/experimental/v1/teams.proto:126]: In other words, the caller does:
```
response, err := client.DeleteTeamMember(...)
if err != nil {
  ... 
}
// got success
```

**Review by ArthurSens — APPROVED** [review]

**Pull request merged**